tests/operators/test_agnostic_load_file.py

```diff
@@ -466,47 +468,94 @@ def test_aql_gcs_file_to_postgres(self):
             warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),
         )
-
-        # Drop target table
         hook.run(
-            f"DROP TABLE IF EXISTS {os.getenv('SNOWFLAKE_SCHEMA')}.{self.SNOWFLAKE_OUTPUT_TABLE_NAME}"
+            f"DROP TABLE IF EXISTS {os.getenv('SNOWFLAKE_SCHEMA')}.{OUTPUT_TABLE_NAME}"
         )
-        self.create_and_run_task(
-            load_file,
-            (),
-            {
-                "path": str(self.cwd) + "/../data/homes.csv",
-                "file_conn_id": "",
-                "output_table": Table(
-                    table_name=self.SNOWFLAKE_OUTPUT_TABLE_NAME,
-                    database=os.getenv("SNOWFLAKE_DATABASE"),
-                    schema=os.getenv("SNOWFLAKE_SCHEMA"),
-                    warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),
-                    conn_id="snowflake_conn",
-                ),
-            },
-        )
-
-        # Read table from db
-        df = hook.get_pandas_df(
-            f"SELECT * FROM {os.getenv('SNOWFLAKE_SCHEMA')}.{self.SNOWFLAKE_OUTPUT_TABLE_NAME}"
+        yield hook
+        hook.run(
+            f"DROP TABLE IF EXISTS {os.getenv('SNOWFLAKE_SCHEMA')}.{OUTPUT_TABLE_NAME}"
         )
-
-        assert df.iloc[0].to_dict() == {
-            "SELL": 142.0,
-            "LIST": 160.0,
-            "LIVING": 28.0,
-            "ROOMS": 10.0,
-            "BEDS": 5.0,
-            "BATHS": 3.0,
-            "AGE": 60.0,
-            "ACRES": 0.28,
-            "TAXES": 3167.0,
+    elif request.param == "postgres":
+        POSTGRES_TABLE_NAME = "test_dag_load_file_homes_csv_2"
+        hook = TempPostgresHook(postgres_conn_id="postgres_conn", schema="pagila")
+        hook.run(f"DROP TABLE IF EXISTS tmp_astro.{OUTPUT_TABLE_NAME}")
+        yield hook
+        hook.run(f"DROP TABLE IF EXISTS tmp_astro.{OUTPUT_TABLE_NAME}")
+    else:
+        raise ValueError(f"Unsupported database {request.param}")
+
+
+@pytest.mark.parametrize("hook", ["snowflake", "postgres"], indirect=True)
+@pytest.mark.parametrize("file_type", ["ndjson", "json", "csv"])
+def test_load_file(sample_dag, hook, file_type):
+    if isinstance(hook, TempPostgresHook):
+        params = {
+            "path": str(CWD) + f"/../data/sample.{file_type}",
+            "file_conn_id": "",
+            "output_table": Table(
+                table_name=OUTPUT_TABLE_NAME,
+                database="pagila",
+                conn_id="postgres_conn",
+            ),
+        }
+        schema = "tmp_astro"
+    else:
+        params = {
+            "path": str(CWD) + f"/../data/sample.{file_type}",
+            "file_conn_id": "",
+            "output_table": Table(
+                table_name=OUTPUT_TABLE_NAME,
+                database=os.getenv("SNOWFLAKE_DATABASE"),
+                schema=OUTPUT_SCHEMA,
+                warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),
+                conn_id="snowflake_conn",
+            ),
         }
+        schema = OUTPUT_SCHEMA
```

Comment:
There seems to be a lot of repeated code here. Are we saving much if we still have to separate the logic in the function on a per db basis?

Reply:
Excellent point, @dimberman; I just tried to improve the repetition - please, let me know your thoughts on the latest version!